src/Symfony/Component/String/BinaryString.php

```diff
@@ -0,0 +1,409 @@
+                        throw new RuntimeException('Matching failed with '.$k.'.');
+                    }
+                }
+
+                throw new RuntimeException('Matching failed with unknown error code.');
+            }
+        } finally {
+            restore_error_handler();
+        }
+
+        return $ret ? $matches : null;
+    }
+
+    public function padBoth(int $length, string $padStr = ' '): parent
+    {
+        $str = clone $this;
+        $str->string = str_pad($this->string, $length, $padStr, STR_PAD_BOTH);
+
+        return $str;
+    }
+
+    public function padEnd(int $length, string $padStr = ' '): parent
+    {
+        $str = clone $this;
+        $str->string = str_pad($this->string, $length, $padStr, STR_PAD_RIGHT);
+
+        return $str;
+    }
+
+    public function padStart(int $length, string $padStr = ' '): parent
+    {
+        $str = clone $this;
+        $str->string = str_pad($this->string, $length, $padStr, STR_PAD_LEFT);
+
+        return $str;
+    }
+
+    public function prepend(string ...$prefix): parent
+    {
+        $str = clone $this;
+        $str->string = (1 >= \count($prefix) ? $prefix[0] ?? '' : implode('', $prefix)).$str->string;
+
+        return $str;
+    }
+
+    public function replace(string $from, string $to): parent
+    {
+        $str = clone $this;
+        $str->string = $this->ignoreCase ? str_ireplace($from, $to, $this->string) : str_replace($from, $to, $this->string);
+
+        return $str;
+    }
+
+    public function replaceMatches(string $from, $to): parent
+    {
+        if ('g' === $pattern[-1]) {
+            $pattern = substr($pattern, 0, -1);
+        }
+
+        if ($this->ignoreCase) {
+            $pattern .= 'i';
+        }
+
+        $replace = \is_array($to) || $to instanceof \Closure ? 'preg_replace_callback' : 'preg_replace';
+        set_error_handler(static function ($t, $m) { throw new InvalidArgumentException($m); });
+
+        try {
+            if (null === $string = $replace($from, $to, $this->string)) {
+                $lastError = preg_last_error();
+
+                foreach (get_defined_constants(true)['pcre'] as $k => $v) {
+                    if ($lastError === $v && '_ERROR' === substr($k, -6)) {
+                        throw new RuntimeException('Matching failed with '.$k.'.');
+                    }
+                }
+
+                throw new RuntimeException('Matching failed with unknown error code.');
+            }
+        } finally {
+            restore_error_handler();
+        }
+
+        $str = clone $this;
+        $str->string = $string;
+
+        return $str;
+    }
+
+    public function slice(int $start = 0, int $length = null): parent
+    {
+        $str = clone $this;
+        $str->string = (string) substr($this->string, $start, $length ?? PHP_INT_MAX);
+
+        return $str;
+    }
+
+    public function snake(): parent
+    {
+        $str = $this->toCamelCase();
```

Comment:
```suggestion
        $str = $this->camel();
```

Reply:
This :camel:? It's not a :desert: :wink:

Reply:
`toCamelCase` does not exist, the method is named `camel`. So it is bug.

Reply:
@gharlan I started reading this PR when it was `toCamelCase()` and didn't see @nicolas-grekas' comment about the change (which I don't like, as a side note). I did not consider that method name was changed and old usage remained.. My bad :)